analyses/malawi/dryspells_trigger/exploration/compare_arc2_chirps.Rmd

```diff
@@ -0,0 +1,483 @@
+for both the centroids and touching method, only a single one of the
+20 and 18 dry spells respectively overlapped with a CHIRPS identified dry spell, and
+none had a dry spell confirmation in January or February. Thus, I explicitly filter
+these out prior to comparing the various dry spell identification methods.
+
+```{r loading}
+# data loading
+
+arc2_center_df <- read_csv(paste0(expl_dir, "/arc2/mwi_arc2_centroid_dry_spells_during_rainy_season.csv"),
+                           col_types = "_ccDDDdd") %>%
+  select(pcode,
+         ADM2_EN,
+         arc2_center_start = dry_spell_first_date,
+         arc2_center_confirmation = dry_spell_confirmation,
+         arc2_center_end = dry_spell_last_date,
+         arc2_center_duration = dry_spell_duration) %>%
+  filter(arc2_center_duration <= 30)
+
+arc2_touch_df <- read_csv(paste0(expl_dir, "/arc2/mwi_arc2_touching_dry_spells_during_rainy_season.csv"),
+                          col_types = "_ccDDDd") %>%
+  select(pcode,
+         ADM2_EN,
+         arc2_touch_start = dry_spell_first_date,
+         arc2_touch_confirmation = dry_spell_confirmation,
+         arc2_touch_end = dry_spell_last_date,
+         arc2_touch_duration = dry_spell_duration) %>%
+  filter(arc2_touch_duration <= 30)
+
+chirps_df <- read_csv(paste0(
+  process_dir, 
+  "/dry_spells/v1/archive/dry_spells_during_rainy_season_list_2000_2020_mean_back.csv"
+)) %>%
+  transmute(pcode,
+            ADM2_EN,
+            chirps_start = dry_spell_first_date,
+            chirps_confirmation = dry_spell_first_date + 13,
+            chirps_end = dry_spell_last_date,
+            chirps_duration = dry_spell_duration)
+
+# shapefile loading
+adm0 <- st_read(paste0(shapefile_dir, "/mwi_admbnda_adm0_nso_20181016.shp"))
+adm1 <- st_read(paste0(shapefile_dir, "/mwi_admbnda_adm1_nso_20181016.shp"))
+adm2 <- st_read(paste0(shapefile_dir, "/mwi_admbnda_adm2_nso_20181016.shp"))
+
+# daily values loading
+arc2_daily_df <- read_csv(paste0(expl_dir, "/arc2/mwi_arc2_precip_long_raw.csv"),
+                          col_types = "____c_______D__dd______") %>%
+  rename(pcode = ADM2_PCODE,
+         arc2_centroid = mean_cell,
+         arc2_touch = mean_cell_touched)
+
+chirps_daily_df <- readRDS(paste0(process_dir, "/dry_spells/v1/data_long_mean_values_2000_2021.RDS")) %>%
+  ungroup() %>%
+  select(pcode, date, chirps = total_prec)
+
+daily_df <- inner_join(arc2_daily_df,
+                       chirps_daily_df,
+                       by = c("pcode", "date"))
+```
+
+With the 3 separate datasets loaded, we can use the `fuzzyjoin` package to join the
+datasets together (with some repititive filtering after) to capture all overlapping
+dry spells to examine in detail how the 3 separate methods for dry spell identification
+compare for each incident in particular.
+
+```{r wrangling}
+# joining ARC2 touch and center dry spells to get all overlaps
+
+arc2_center_lj1 <- fuzzy_full_join(
```

Comment:
Essentially! Gets messy because joining 3 data frames together so might be ways to clean it up but couldn't figure out a cleaner way in such a short time frame.